src/pybind/mgr/prometheus/module.py

```diff
@@ -450,6 +460,15 @@ def get_pool_stats(self):
                     pool['recovery_rate'].get(stat, 0),
                     (pool['pool_id'],)
                 )
+    def get_osd_df(self):
+        osd_df = self.get('osd_df')
+        for osd in osd_df['nodes']:
+            name_ = osd['name']
```

Comment:
why do you want to get the *last* `osd['name']`?

Reply:
why not using `name` for the variable name?

Reply:
> why not using `name` for the variable name?

> name_ is used in order to diff the osd[`name`].

Reply:
> why do you want to get the _last_ `osd['name']`?

> no, i want to get all osd name by the `for` loop.

Reply:
> > why do you want to get the _last_ `osd['name']`?
> 
> no, i want to get all osd name by the `for` loop.

Then this code does not do what you expect it to do.